### PR TITLE
Add Support for the Avnet MaaXBoard

### PR DESCRIPTION
**Pull request by jdselby-cap**

### CODE REVIEW

File: src/plat/maaxboard/config.cmake [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#
+# Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+# Copyright 2022, Capgemini Engineering
```
Comment (jdselby-cap) [inline src/plat/maaxboard/config.cmake:3]: Does this need to be removed?

File: src/plat/maaxboard/overlay-maaxboard-32bit.dts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+ * Copyright 2022, Capgemini Engineering
```
Comment (jdselby-cap) [inline src/plat/maaxboard/overlay-maaxboard-32bit.dts:3]: Does this need to be removed?

File: src/plat/maaxboard/overlay-maaxboard.dts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		reg = < 0x00 0x302e0000 0x00 0x10000 >;
+		interrupts = < 0x00 0x36 0x04 >;
+		clocks = < 0x04 0xc5 0x04 0xc5 0x04 0xf9 >;
+		clock-names = "ipg\0per\0osc_per";
+		status = "disabled";
+	};
+   
+	gpt@302f0000 {
+		compatible = "fsl,imx8mq-gpt\0fsl,imx7d-gpt";
+		reg = < 0x00 0x302e0000 0x00 0x10000 >;
```
Comment (mjenkcapgemini) [inline src/plat/maaxboard/overlay-maaxboard.dts:41]: Should this read < ... 0x302f0000 ... > to match the earlier line?

File: src/plat/maaxboard/overlay-maaxboard.dts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+		interrupts = < 0x00 0x35 0x04 >;
+		clocks = < 0x04 0xc5 0x04 0xc5 0x04 0xf9 >;
+		clock-names = "ipg\0per\0osc_per";
+		status = "disabled";
+	};
+
+	gpt@302f0000 {
```
Comment (mjenkcapgemini) [inline src/plat/maaxboard/overlay-maaxboard.dts:48]: This appears to be a duplicate of the preceding gpt@ {}
Should it be at a different address?

File: src/plat/maaxboard/overlay-maaxboard.dts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+		compatible = "fsl,imx8mq-gpt\0fsl,imx7d-gpt";
+		reg = < 0x00 0x302e0000 0x00 0x10000 >;
```
Comment (mjenkcapgemini) [inline src/plat/maaxboard/overlay-maaxboard.dts:50]: Should < ... 0x302e0000 ... > match the earlier line (modified if necessary for the preceding comment)?

File: src/plat/maaxboard/overlay-maaxboard.dts [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+	};
+
+
+
+   
+         IRQ
+   gpt1  55   302D_0000
+   gpt2  54   302E_0000
+   gpt3  53   302F_0000
+   gpt4  52   3070_0000
```
Comment (mjenkcapgemini) [inline src/plat/maaxboard/overlay-maaxboard.dts:64]: If the previous gpt@ {} comments result in a change, will it impact this table?

File: src/plat/maaxboard/overlay-maaxboard.dts [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+		reg = < 0x00 0x302e0000 0x00 0x10000 >;
+		interrupts = < 0x00 0x34 0x04 >;
+		clocks = < 0x04 0xc5 0x04 0xc5 0x04 0xf9 >;
+		clock-names = "ipg\0per\0osc_per";
+		status = "disabled";
+	};
+
+	gpt@306f0000 {
+		compatible = "fsl,imx8mq-gpt\0fsl,imx7d-gpt";
+		reg = < 0x00 0x302e0000 0x00 0x10000 >;
```
Comment (mjenkcapgemini) [inline src/plat/maaxboard/overlay-maaxboard.dts:59]: Should this read < ... 0x306f0000 ... > to match the earlier line?

File: src/plat/maaxboard/overlay-maaxboard.dts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+		interrupts = < 0x00 0x33 0x04 >;
+		clocks = < 0x04 0xc5 0x04 0xc5 0x04 0xf9 >;
+		clock-names = "ipg\0per\0osc_per";
+		status = "disabled";
+	};
+
+	gpt@306e0000 {
+		compatible = "fsl,imx8mq-gpt\0fsl,imx7d-gpt";
+		reg = < 0x00 0x302e0000 0x00 0x10000 >;
+		interrupts = < 0x00 0x2e 0x04 >;
```
Comment (mjenkcapgemini) [inline src/plat/maaxboard/overlay-maaxboard.dts:69]: Where can I look this up to check? In terms of pattern, the second item in interrupts = < ... ... ... > has been steadily decrementing by 1 for each successive GPT entry. Solely on that basis, I expected 0x32 here. 0x2e is probably right but I feel I should check.

Comment (mjenkcapgemini) [reply src/plat/maaxboard/overlay-maaxboard.dts:69]: Answer: Table 7-1 of NXP imx8m reference manual.
This checks out, the irq numbering is just oddly discontiguous, but correct.
Resolved.

File: tools/dts/maaxboard.dts [hunk 9]
```diff
@@ -0,0 +1,1846 @@
...
+ * The licenses of all input files to this process are compatible
+ * with GPL-2.0.
+ *
+ * The following commands were used to derive this file:
+ * > git clone https://github.com/Avnet/linux-imx.git
+ * > cd linux-imx
+ * > cpp -nostdinc -I include -I arch -undef -x assembler-with-cpp \
+ * >    arch/arm64/boot/dts/freescale/maaxboard-base.dts temp.dts
+ * > dtc temp.dts temp.dtb
+ * > dtc -I dtb -O dts temp.dtb > maaxboard.dts
```
Comment (mjenkcapgemini) [inline tools/dts/maaxboard.dts:30]: Following on from the aforementioned documentation comment, although this line works, it is equivalent to the following line, which more closely mirrors the temp.dts -> temp.dtb translation and has been preferred in the updated documentation:
`dtc -o maaxboard.dts temp.dtb`

Comment (mjenkcapgemini) [inline tools/dts/maaxboard.dts:30]: Following on from the aforementioned documentation comment, although this line works, it is equivalent to the following line, which more closely mirrors the temp.dts -> temp.dtb translation and has been preferred in the updated documentation:
`dtc -o maaxboard.dts temp.dtb`

Comment (mjenkcapgemini) [reply tools/dts/maaxboard.dts]: Updated as suggested.

File: src/plat/maaxboard/overlay-maaxboard.dts (hunk 3 above)
Comment (mjenkcapgemini) [reply src/plat/maaxboard/overlay-maaxboard.dts]: Agreed with Stephen and fixed.

File: src/plat/maaxboard/overlay-maaxboard.dts (hunk 5 above)
Comment (mjenkcapgemini) [reply src/plat/maaxboard/overlay-maaxboard.dts]: Agreed with Stephen and fixed.

File: src/plat/maaxboard/overlay-maaxboard.dts (hunk 7 above)
Comment (mjenkcapgemini) [reply src/plat/maaxboard/overlay-maaxboard.dts]: Agreed with Stephen and fixed.

File: src/plat/maaxboard/overlay-maaxboard.dts [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+		reg = < 0x00 0x302e0000 0x00 0x10000 >;
+		interrupts = < 0x00 0x33 0x04 >;
+		clocks = < 0x04 0xc5 0x04 0xc5 0x04 0xf9 >;
+		clock-names = "ipg\0per\0osc_per";
+		status = "disabled";
+	};
+
+	gpt@306e0000 {
+		compatible = "fsl,imx8mq-gpt\0fsl,imx7d-gpt";
+		reg = < 0x00 0x302e0000 0x00 0x10000 >;
```
[the remark this answers is not in the document]
Comment (mjenkcapgemini) [reply src/plat/maaxboard/overlay-maaxboard.dts]: Agreed with Stephen and fixed.

**Comment by mjenkcapgemini** [comment]
Housekeeping: this PR can now be closed without merging. Any merges are superseded by our seL4 Foundation PRs.

**Pull request closed without merging**